xarray/plot/plot.py

```diff
@@ -449,6 +450,21 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
         if imshow_rgb:
             # Don't add a colorbar when showing an image with explicit colors
             add_colorbar = False
+            # Calculate vmin and vmax automatically for `robust=True`
+            if robust:
+                vmin_tmp, vmax_tmp = darray.quantile(
+                    q=[ROBUST_PERCENTILE, 1 - ROBUST_PERCENTILE])
+                vmin = vmin if vmin is not None else vmin_tmp
+                vmax = vmax if vmax is not None else vmax_tmp
+                del vmin_tmp, vmax__tmp
+                robust=False
```

Comment:
2 nits: 1) let garbage collection handle deleting your temp variables, 2) space in `robust = False`

Reply:
I think we use `locals()` somewhere below, which is why this is necessary (but yes it's pretty ugly)